### PR TITLE
Implement load balancing loss + fix some bugs in ExpertBiasTracker

### PR DESCRIPTION
**Pull request by bltnynk**
WIP
- [x] Implement load balancing loss
- [ ] Fix bug in `routed_forward` and `ExpertBiasTracker`.

### CODE REVIEW

File: src/levanter/models/routed_qwen_model.py [hunk 1]
```diff
@@ -747,28 +750,28 @@ def init(*args, **kwargs):
         linear = hnn.Linear.init(*args, **kwargs)
         return Router(linear.weight, linear.bias, linear.In, linear.Out, linear.dot_general)
 
-
-class ExpertBiasTracker(eqx.Module):
-    curr_bias: NamedArray
+@dataclass
+class ExpertBiasTracker():
+    bias: NamedArray
     load: NamedArray
```
Comment (bltnynk) [inline src/levanter/models/routed_qwen_model.py]: Btw, what's the point of saving `load` inside this `ExpertBiasTracker`?
As far as I concern we should only compute the load within an example of a Batch, due to microbatching,. Hence, there's really no need to aggregate the load for entire batch, right?
(Except if we do logging, then we need this. Otherwise I don't really see its use).

Comment (Aphoh) [reply src/levanter/models/routed_qwen_model.py]: We can't modify anything in the `ExpertBiasTracker` within the model, so it stores whatever the previous bias was, and also the expert load for the last batch. Then it computes the correct bias for this iteration, and computes the new load for this iteration. The loads get aggregated in the `__add__` method which is used by the microbatching logic, and it sets the new bias to be the computed one from this iteration.

File: src/levanter/models/routed_qwen_model.py [hunk 2]
```diff
@@ -747,28 +750,28 @@ def init(*args, **kwargs):
         linear = hnn.Linear.init(*args, **kwargs)
         return Router(linear.weight, linear.bias, linear.In, linear.Out, linear.dot_general)
 
-
-class ExpertBiasTracker(eqx.Module):
-    curr_bias: NamedArray
```
Comment (Aphoh) [inline src/levanter/models/routed_qwen_model.py]: We should actually rename this to `prev_bias`

File: src/levanter/models/routed_qwen_model.py [hunk 3]
```diff
@@ -747,28 +750,28 @@ def init(*args, **kwargs):
...
+    bias: NamedArray
     load: NamedArray
 
-    def __add__(self, other: "ExpertBiasTracker") -> "ExpertBiasTracker":
-        return ExpertBiasTracker(self.curr_bias, self.load + other.load)
-
     @staticmethod
-    def zero(config: RQwenConfig):
-        return ExpertBiasTracker(hax.zeros(config.Experts), hax.zeros(config.Experts))
+    def init(config: RQwenConfig):
```
Comment (Aphoh) [inline src/levanter/models/routed_qwen_model.py]: Why change the name to init? `zero` is much clearer.

File: src/levanter/models/routed_qwen_model.py (hunk 1 above)
Comment (Aphoh) [inline src/levanter/models/routed_qwen_model.py]: Likewise this should be `prev_load`

File: src/levanter/models/routed_qwen_model.py [hunk 4]
```diff
@@ -949,6 +955,13 @@ def routed_forward(
         else:
             expert_mask = hax.where(router_hs_idxs < 0, 0.0, expert_mask).astype(compute_dtype)
 
+        # Compute the number of tokens routed to each experts.
+        used_experts = expert_mask > 0
+        # Since we are routing by sequence, this step is necessary.
+        used_experts /= example.seq_length
+        # Due to microbatchibg, we sum across (Pos, ) instead of (Batch, Pos)
+        expert_load = hax.sum(used_experts, (Pos,), where=example.completion_mask, dtype=compute_dtype)
```
Comment (Aphoh) [inline src/levanter/models/routed_qwen_model.py]: This isn't correct, let's circle back on this

**Review by Aphoh — CHANGES_REQUESTED** [review]

File: src/levanter/models/routed_qwen_model.py (hunk 3 above)
Comment (bltnynk) [reply src/levanter/models/routed_qwen_model.py]: Fixed

File: src/levanter/models/routed_qwen_model.py (hunk 2 above)
Comment (bltnynk) [reply src/levanter/models/routed_qwen_model.py]: I think since the class and method is functional, every method return a new `ExpertBiasTracker`, we can just leave it as `bias` and `load`, which looks nicer.

File: src/levanter/models/routed_qwen_model.py (hunk 4 above)
Comment (bltnynk) [reply src/levanter/models/routed_qwen_model.py]: Fixed (not really sure though)

File: src/levanter/models/routed_qwen_model.py [hunk 5]
```diff
@@ -82,11 +82,14 @@ def __call__(self, x: NamedArray, expert_mask: Optional[NamedArray], *, key=None
 
 
 def is_routed_experts_param(x):
-    return isinstance(x, (RQwenMlpExperts, Router, LowRankLinear))
```
Comment (Aphoh) [inline src/levanter/models/routed_qwen_model.py]: Oh dude this means you won't be training the router

**Review by Aphoh — CHANGES_REQUESTED** [review]
